src/api/app/policies/report_policy.rb

```diff
@@ -0,0 +1,28 @@
+class ReportPolicy < ApplicationPolicy
+  attr_reader :user, :record
+
+  def initialize(user, record)
+    super(user, record)
+
+    @user = user
+    @record = record
+  end
+
+  def show?
+    user.is_admin? || record.user == user
+  end
+
+  def submit_report?
+    not_self = if record.instance_of?(::Comment)
+                 record.user != user
+               elsif record.instance_of?(::User)
+                 record != user
+               elsif record.instance_of?(::Package)
+                 record.project.users.exclude?(user)
```

Comment:
This should be limited to the maintainer role, same for project.

Reply:
Do you mean maintainers should be able to report the projects?

Reply:
Ah sorry I've read `include?`....

Why would a a user of the package/project not be able to report it?

Reply:
Same reason as `comments` and `users`. I think it's not logical to add a comment and then report it yourself Or report your own `user` profile.

Reply:
For comments and users sure, but *inside* a group of people that work together (on a package or project) there can be abuse for instance.

Reply:
That makes sense. I got it :+1:

Reply:
I've updated this policy. Please check again